File: server/src/main/java/com/tnt/food_delivery/repository/UserRepository.java
package com.tnt.food_delivery.repository;


import com.tnt.food_delivery.model.User;
import com.tnt.food_delivery.model.request.AuthenticationRequestEntity;

import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.data.mongodb.repository.Query;

import java.util.List;

public interface UserRepository extends MongoRepository<User, String> {
    @Query("{ name: { $regex: ?0, $options: 'i' }, userRole: RESTAURANT}")
    List<User> findRestaurantByName(String name);

    @Query("{ '$or':[ { 'username': ?0 , 'password': ?1 }, { 'email': ?0 , 'password': ?1 }, { 'phoneNumber': ?0 , 'password': ?1 } ] }")
    User login(String username, String password);

    @Query("{'$or': [ {'username': ?0} , {'email': ?1} ] }")
    User checkRegister(String username, String email);
}

File: native/app/src/main/java/com/tnt/food_delivery/network/repository/UserRepository.kt
package com.tnt.food_delivery.network.repository

import com.tnt.food_delivery.core.Action
import com.tnt.food_delivery.data.model.User
import kotlinx.coroutines.flow.Flow

interface UserRepository {
    fun loginWith(login: String, password: String): Flow<Action<User>>

    fun registerWith(
        name: String,
        surname: String,
        nickname: String,
        email: String,
        password: String
    ): Flow<Action<User>>
}